services/motion/builtin/move_request.go

```diff
@@ -177,33 +181,158 @@ func (ms *builtIn) newMoveOnGlobeRequest(
+func (ms *builtIn) newMoveOnMapRequest(
+	ctx context.Context,
+	componentName resource.Name,
+	goalPoseRaw spatialmath.Pose,
+	slamName resource.Name,
+	kinematicsOptions kinematicbase.Options,
+	extra map[string]interface{},
+) (*moveRequest, error) {
+	// get the SLAM Service from the slamName
+	slamSvc, ok := ms.slamServices[slamName]
+	if !ok {
+		return nil, resource.DependencyNotFoundError(slamName)
+	}
+
+	// gets the extents of the SLAM map
+	limits, err := slam.Limits(ctx, slamSvc)
+	if err != nil {
+		return nil, err
+	}
+	limits = append(limits, referenceframe.Limit{Min: -2 * math.Pi, Max: 2 * math.Pi})
+
+	// create a KinematicBase from the componentName
+	component, ok := ms.components[componentName]
+	if !ok {
+		return nil, resource.DependencyNotFoundError(componentName)
+	}
+	b, ok := component.(base.Base)
+	if !ok {
+		return nil, fmt.Errorf("cannot move component of type %T because it is not a Base", component)
+	}
+
+	if extra != nil {
+		if profile, ok := extra["motion_profile"]; ok {
+			motionProfile, ok := profile.(string)
+			if !ok {
+				return nil, errors.New("could not interpret motion_profile field as string")
+			}
+			kinematicsOptions.PositionOnlyMode = motionProfile == motionplan.PositionOnlyMotionProfile
+		}
+	}
+
+	fs, err := ms.fsService.FrameSystem(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	kb, err := kinematicbase.WrapWithKinematics(ctx, b, ms.logger, motion.NewSLAMLocalizer(slamSvc), limits, kinematicsOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// get point cloud data in the form of bytes from pcd
+	pointCloudData, err := slam.PointCloudMapFull(ctx, slamSvc)
+	if err != nil {
+		return nil, err
+	}
+	// store slam point cloud data  in the form of a recursive octree for collision checking
+	octree, err := pointcloud.ReadPCDToBasicOctree(bytes.NewReader(pointCloudData))
+	if err != nil {
+		return nil, err
+	}
+
+	mr, err := relativeMoveRequestFromAbsolute(
+		ctx,
+		nil,
+		ms.logger,
+		kb,
+		goalPoseRaw,
+		fs,
+		[]spatialmath.Geometry{octree},
+		extra,
+	)
+	return mr, err
+}
+
+func relativeMoveRequestFromAbsolute(
+	ctx context.Context,
+	motionCfg *motion.MotionConfiguration,
+	logger golog.Logger,
+	kb kinematicbase.KinematicBase,
+	goalPoseInWorld spatialmath.Pose,
+	fs referenceframe.FrameSystem,
+	worldObstacles []spatialmath.Geometry,
+	extra map[string]interface{},
+) (*moveRequest, error) {
 	// replace original base frame with one that knows how to move itself and allow planning for
 	kinematicFrame := kb.Kinematics()
-	if err = fs.ReplaceFrame(kinematicFrame); err != nil {
+	if err := fs.ReplaceFrame(kinematicFrame); err != nil {
 		return nil, err
 	}
 	// We want to disregard anything in the FS whose eventual parent is not the base, because we don't know where it is.
 	baseOnlyFS, err := fs.FrameSystemSubset(kinematicFrame)
 	if err != nil {
 		return nil, err
 	}
-	startPose, err := kb.CurrentPosition(ctx)
+
+	startPoseInWorld, err := kb.CurrentPosition(ctx)
```

Comment:
I'd rather call it just startPose because its not guaranteed this will be World in the future

Reply:
True